### ISSUE TITLE
Lack of editing privileges when I am the submitter of a dataset

### ISSUE THREAD

**Issue by clarallebot** [osulp/Scholars-Archive#1971]
### Descriptive summary

I am an administrator for datasets, I review them, and I approve them. Occasionally a user has exceptional circumstances that require that I create the dataset for them (i.e. they don't have an ONID). When I do that, I am at the same time the author and the reviewer. ScholarsArchive@OSU seems to prioritize my role as an author over my role as a reviewer, and I loose editing privileges that I have as a reviewer. I would like ScholarsArchive@OSU to prioritize my role as administrator so that I can edit the entry as a reviewer, and not just as an author.   

### Expected behavior

ScholarsArchive@OSU recognizes that I am an authorized admin for any datasets, regardless of whether I own them or not. This means that I can:
* Request a change or approve the dataset by clicking under "Review and approval"
* Edit, attach child, or delete the record, as shown by the buttons on the top right, next to the title.  

### Actual behavior

In datasets I do not own I am authorized to make changes to a dataset at any time, regardless of the status of the dataset. I can Request a change or approve the dataset by clicking under "Review and approval" and I can Edit, attach child, or delete the record, as shown by the buttons on the top right, next to the title. 

In datasets I own I can Request a change or approve the dataset by clicking under "Review and approval". However, I do not see the buttons of Edit, attach child, or delete, and therefore I cannot edit, attach child or delete the dataset. I only see the Analytics and Add to collection buttons. 

If I "Request changes" then I see the Edit, Delete and Attach as child button, as any author would see, and I can make changes as an author. 

### Steps to reproduce the behavior

Not sure if this happens to all admins, or just me. This are the steps I need to do to reproduce the problem:
* Create a dataset: Go to Share your work, choose dataset, fill the fields, attach documents and check the publishing agreement. Click on Save. I get an e-mail as the author, saying "Thank you for your deposit to ScholarsArchive@OSU". I get an e-mail as an admin saying "Deposit needs review". 
* I go to the dataset, and I cannot see the Edit, Delete and Attach as child button. 

### Related work

Right now it is happening with this dataset I own: https://ir.library.oregonstate.edu/concern/datasets/1r66j709p

**Comment by straleyb** [comment]
This is due to works needing to go through the review process. When a work is under review, it needs changes requested before it can be edited again. If thats not what we would like to happen, and being an admin overrides the workflow functionality, then that can be changed. Which would we like?

**Comment by clarallebot** [comment]
I got an alert that this message had been submitted, but I cannot find it "This is due to works needing to go through the review process. When a work is under review, it needs changes requested before it can be edited again. If thats not what we would like to happen, and being an admin overrides the workflow functionality, then that can be changed. Which would we like?"

In any case, not sure I understand the question. What are the two options?

**Comment by straleyb** [comment]
I think we decided on not worrying about pushing it through the review process. 

`As an admin, I should be able to edit a work that is currently in the review process`

Im not sure where that message came from. But i'll be working on this today.

**Comment by clarallebot** [comment]
Sounds good, thanks!

**Comment by simholt** [comment]
@straleyb 
Thanks to @shieldsb for help testing this one!
***
I made Beth a Dataset Manager. She was able to upload a dataset and edit it while it was in the review queue and after it had been approved. All good there.

![image__2_](https://user-images.githubusercontent.com/33705514/75932315-f5e12880-5e2b-11ea-8b9f-e0ec93e519c2.png)
***
I then created a dataset, transferred it to Beth, and she was able to make edits. 

QUESTION: What is up with the Request Deletion button in the image above? I haven't seen that before. Does that only show for Managers?

QA pass --
